### PR TITLE
Export config file & data as zip file

### PR DESCRIPTION
**Pull request by devenyantis**
# Description

User is able to choose whether to

- export config file only
- export config file with data (zip together)

**Update [Refer this commit](https://github.com/CertifaiAI/classifai/pull/345/commits/560003e855a81a32033fd4c9625537f24d9bf375)**
To solve blocking thread problem, choosing export type should be done from frontend.
`http://localhost:9999/v2/:annotation_type/projects/:project_name/export/:export_type`

export_type
- `cfg` To export config file only
- `cfgdata` To export config file + data. Will produce a zip file

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Tested on?

- [X] Windows  
- [ ] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged

### CODE REVIEW

**Comment by codenamewei** [comment]
A few thing I realised when runnning it

- Screenshot 2021-04-13 at 5 19 06 PM

Capital should be better for consistency 

- https://gist.github.com/codenamewei/76a5aaa2cfdf3d3d0281f74fe310582d

Thread is blocked when the above window stay opens for 15 seconds and above

However the same scenario did not happen when the project folder stay open ...
Screenshot 2021-04-13 at 5 22 08 PM

or even this pop up message below stay open for more than 15 seconds
Screenshot 2021-04-13 at 5 32 38 PM



as much of it doesnt causing error for now , would like to relook into to 
(1) not possibly blocking the thread and consuming computation power
(2) spamming the log file which hinders effective debugging

@devenyantis

File: classifai-core/src/main/java/ai/classifai/action/ActionOps.java [hunk 1]
```diff
@@ -144,4 +145,19 @@ private static String removePair(@NonNull String input, char openBracket, char c
 
         return input.substring(1, input.length() - 1);
     }
+
+    public static ExportType getExportType(String export_type_str)
```
Comment (codenamewei) [inline classifai-core/src/main/java/ai/classifai/action/ActionOps.java:149]: export_type_str to CamelCase thanks

**Review by codenamewei — CHANGES_REQUESTED** [review]

**Comment by devenyantis** [comment]
@codenamewei 

Merged PR for choose export options

Capture1

**Comment by codenamewei** [comment]
@amalsyahmi-certifai @devenyantis 
The layout need changes ....
i find the **cancel** button come first is wacky

suggest layout like below..

for those who decide not to perform any operation can click close button to exit

The button **Config File** & **Zip File (With Data)** have to be same size too
hence the two layouts for suggestion
alternatively the phrase **(With Data)** can be made with a smaller font too (because its of extra note)

Image from iOS (2)
Image from iOS (1)

**Review by codenamewei — CHANGES_REQUESTED** [review]
As per comment

**Comment by devenyantis** [comment]
> @amalsyahmi-certifai @devenyantis
> The layout need changes ....
> i find the **cancel** button come first is wacky
> 
> suggest layout like below..
> 
> for those who decide not to perform any operation can click close button to exit
> 
> The button **Config File** & **Zip File (With Data)** have to be same size too
> hence the two layouts for suggestion
> alternatively the phrase **(With Data)** can be made with a smaller font too (because its of extra note)
> 
> Image from iOS (2)
> Image from iOS (1)

@amalsyahmi-certifai I'm fine with the first one. So, the size won't be too big. 
And maybe the `close` button can be smaller than the options.

**Comment by codenamewei** [comment]
> @amalsyahmi-certifai I'm fine with the first one. So, the size won't be too big.
> And maybe the `close` button can be smaller than the options.

yes im onboard on the smaller close button too...thanks guys

**Comment by amalsyahmi-certifai** [comment]
@codenamewei @devenyantis , is this good?
image

**Comment by amalsyahmi-certifai** [comment]
@codenamewei @devenyantis 

Design proposed by @codenamewei . The 'Config File' text aligns with 'Zip File' in the second button. For me, I like the previous design ('Config File' text aligns center in the button). What do you think?

image

**Comment by BobbyLCW** [comment]
I would like to suggest the cancel button take away. make it a cross (x) on top right corner (size smaller). At the same time, enable click on blank space can exit this interface.

**Comment by codenamewei** [comment]
> I would like to suggest the cancel button take away. make it a cross (x) on top right corner (size smaller). At the same time, enable click on blank space can exit this interface.

@amalsyahmi-certifai Yeah , alignment on first one is better

can you also check out Bobby comments?
the x on the top is align with classifai other interface
Screenshot 2021-04-20 at 2 25 32 PM

, which failed to cross my mind till now


also the click on blank space to exit too, the same thing can implement on this shortcut key page

**Comment by amalsyahmi-certifai** [comment]
@codenamewei @BobbyLCW 
Thank you for the suggestions.
I will proceed with the first design first. For cross on top right corner and click outside of the modal to close, I will do in the next PR.

**Review by codenamewei — APPROVED** [review]
👍

**Pull request merged**